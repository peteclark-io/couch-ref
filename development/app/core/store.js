'use strict';

import { createStore } from 'redux';
import reducer from '../ducks/reducers';

const store = createStore(reducer);

export default store;
