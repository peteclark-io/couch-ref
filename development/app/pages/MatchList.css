.spacer {
  margin-bottom: 2em;
}
