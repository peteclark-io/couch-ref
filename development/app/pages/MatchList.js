'use strict'

import React from 'react';

import styles from './MatchList.css';
import LiveScores from '../components/Scores/LiveScores';
import CouchRefHeader from '../components/Common/CouchRefHeader';

const MatchList = React.createClass({

  render: function() {
    return (
      <div>
         <h2 className={styles.header}>Live Scores</h2>
         <div className={styles.spacer}></div>
         <LiveScores />
      </div>
    );
  }
});

export default MatchList;
