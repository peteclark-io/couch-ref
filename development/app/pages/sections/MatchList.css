@import "../../css/branding.css";

.spacer {
  margin-bottom: 36px;
}

.header {
  font-family: var(--font-primary);
  margin-top: 20px;
}
