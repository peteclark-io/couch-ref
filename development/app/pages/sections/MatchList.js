'use strict'

import React from 'react';
import _ from 'lodash';
import moment from 'moment';
import { connect } from 'react-redux'

import Fixtures from '../../components/MatchScores/Fixtures';
import RecentFixtures from '../../components/MatchScores/RecentFixtures';

export const MatchList = React.createClass({

   propTypes: {
      title: React.PropTypes.string,
      fixtures: React.PropTypes.array,
      recentFixtures: React.PropTypes.array
   },

   render: function() {
      return (
         <div>
            <Fixtures title={this.props.title} fixtures={this.props.fixtures} />
            <RecentFixtures fixtures={this.props.recentFixtures} />
         </div>
      );
   }
});

const getRecentFixtures = (yesterday, state = {matches: {}}) => {
   var fixtures = _.values(state.matches);

   return _.reverse(_.sortBy(_.filter(fixtures, i => {
      return i.kickOff.isBefore(yesterday) && i.televised && i.live;
   }), ['kickOff']));
}

const getFixtures = (today, yesterday, state = {matches: {}}) => {
   var matches = _.values(state.matches);

   var todaysFixtures = _.filter(matches, (i) => {
      return i.kickOff.isBefore(today) && i.kickOff.isAfter(yesterday) && i.televised;
   });

   var title = 'Upcoming Fixtures';
   var fixtures = [];
   if (todaysFixtures.length > 0){
      fixtures = todaysFixtures;
      title = 'Today\'s Fixtures';
   } else {
      fixtures = _.filter(matches, (i) => {
         return i.kickOff.isAfter(today) && i.televised;
      });
   }

   fixtures = _.sortBy(fixtures, ['fullTime', 'kickOff']);
   return {title: title, fixtures: fixtures};
};

const mapStateToProps = (state) => {
   var today = moment(24, 'HH'); // midnight tonight
   var yesterday = moment(24, 'HH').subtract(1, 'days'); // midnight last night
   var fixtures = getFixtures(today, yesterday, state);

   return {
      fixtures: fixtures.fixtures,
      title: fixtures.title,
      recentFixtures: getRecentFixtures(yesterday, state)
   }
};

const LiveMatchList = connect(
   mapStateToProps
)(MatchList);

export default LiveMatchList;
