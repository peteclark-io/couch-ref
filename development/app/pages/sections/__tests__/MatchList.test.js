'use strict';

import React from 'react';
import MatchList from '../MatchList';
import {shallow} from 'enzyme';

import {Provider} from 'react-redux';
import store from '../../../core/store';

test('MatchList section', () => {
   const rendered = shallow(
      <Provider store={store}>
         <MatchList />
      </Provider>
   );

   let tree = rendered.debug();
   expect(tree).toMatchSnapshot();
});
