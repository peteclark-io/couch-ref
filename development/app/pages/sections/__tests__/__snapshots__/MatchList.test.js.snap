exports[`test MatchList section 1`] = `"<MatchList />"`;
